packages/flutter_tools/lib/src/build_system/targets/web.dart

```diff
@@ -405,12 +449,15 @@ const RESOURCES = {
 };
 
 // The application shell files that are downloaded before a service worker can
-// start.
+// start when using offline-first strategy.
 const CORE = [
   ${coreBundle.map((String file) => '"$file"').join(',\n')}];
 
 // During install, the TEMP cache is populated with the application shell files.
 self.addEventListener("install", (event) => {
+  if ('${serviceWorkerStrategy.toString()}' != 'ServiceWorkerStrategy.offlineFirst') {
```

Comment:
Why make this a JS expression? It will effectively compile to `if (true)` and `if (false)`. If put the if outside the string interpolation we could just not emit the `else` branches.